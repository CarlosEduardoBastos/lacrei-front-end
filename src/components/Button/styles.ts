import { NavLink } from 'react-router-dom';
import styled, { css, DefaultTheme } from 'styled-components';

export const Container = styled(NavLink)`
  ${({ theme, background }:{theme: DefaultTheme, background: Number}) => css`
    font-size: 1.125rem;
    width: 192px;
    height: 48px;
    display: flex;
    align-items: center;
    justify-content: center;
    background: ${background ? theme.colors.primaryColor : 'transparent'};
    box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);
    color: ${background ? theme.colors.white : theme.colors.primaryColor};
    border-radius: 8px;
    border: ${background ? 'none' : '2px solid #018762'};
    transition: all 0.2s ease 0s;
    font-weight: 700;
    margin: 10px 0;

    &:hover {
      opacity: 0.7;
    }
  `}
`;
