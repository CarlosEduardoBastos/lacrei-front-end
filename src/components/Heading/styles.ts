import styled, { css, DefaultTheme } from 'styled-components';
import { HeadingProps } from '.';

const titleSize = {
  small: (theme: DefaultTheme) => css`
    font-size: ${theme.fonts.sizes.xsmall};
  `,
  medium: (theme: DefaultTheme) => css`
    font-size: ${theme.fonts.sizes.medium};
  `,
  big: (theme: DefaultTheme) => css`
    font-size: ${theme.fonts.sizes.xxlarge};
    ${mediaFont(theme)}
  `,
};

const mediaFont = (theme: DefaultTheme) => css`
  @media (max-width: 1039px) {
    font-size: ${theme.fonts.sizes.large};
  }
`;

export const Container = styled.h1<HeadingProps>`
  ${({ theme, size }) => css`
    color: ${theme.colors.secondaryColor};
    font-weight: bold;
    ${size ? titleSize[size](theme) : null}
  `}
`;
