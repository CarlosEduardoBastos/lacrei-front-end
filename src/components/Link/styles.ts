/* eslint-disable no-unused-vars */
import { NavLink } from 'react-router-dom';
import styled, { css } from 'styled-components';

const mediaQuery = () => css`
  @media (max-width: 768px) {
    font-size: 14px;
    padding: 0 0.625rem;
  }
`;

export const Container = styled(NavLink)`
  ${({ theme }) => css`
    color: ${theme.colors.secondaryColor};
    font-size: ${theme.fonts.sizes.small};
    font-weight: 700;
    transition: all 0.2s ease 0s;
    padding: 0 1.25rem;
    line-height: 22px;

    &:hover {
      color: ${theme.colors.secondaryColorHover};
    }

    &.active {
      color: ${theme.colors.primaryColor};
    }
    ${mediaQuery()}
  `}
`;
