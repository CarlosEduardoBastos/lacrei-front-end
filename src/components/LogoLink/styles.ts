import { Link } from 'react-router-dom';
import styled, { css } from 'styled-components';

export const Container = styled(Link)`
  ${({ theme }) => css`
    font-size: ${theme.fonts.sizes.large};
    color: ${theme.colors.primaryColor};
    font-weight: 700;
    transition: all 0.2s ease 0s;
    line-height: 44px;

    &:hover {
      color: ${theme.colors.primaryColorHover};
    }
  `}
`;
