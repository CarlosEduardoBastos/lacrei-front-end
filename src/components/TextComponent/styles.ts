import styled, { css, DefaultTheme } from 'styled-components';

const mediaFont = (theme: DefaultTheme) => css`
  @media (max-width: 1039px) {
    font-size: ${theme.fonts.sizes.small};
    margin-top: 1rem;
  }
`;

export const Container = styled.p`
  ${({ theme }) => css`
    color: ${theme.colors.colorText};
    font-size: ${theme.fonts.sizes.medium};
    margin-top: 2rem;
    ${mediaFont(theme)}
  `}
`;
