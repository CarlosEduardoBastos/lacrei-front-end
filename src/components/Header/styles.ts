import styled, { css } from 'styled-components';
import { Container as SectionContainer } from '../SectionContainer/styles';

const mediaQuery = () => css`
  @media (max-width: 520px) {
    justify-content: center;
  }
`;

export const Container = styled.header`
  ${({ theme }) => css`
    width: 100%;
    background: ${theme.colors.colorHeader};
    padding: 0.5rem 4rem;

    ${SectionContainer} {
      padding: 0;
    }
  `}
`;

export const MenuContainer = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
  flex-wrap: wrap;
  ${mediaQuery()}
`;

export const NavLink = styled.ul`
  display: flex;
  align-items: center;
  justify-content: center;
`;
