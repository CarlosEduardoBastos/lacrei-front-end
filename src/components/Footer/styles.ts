
import styled, { css } from 'styled-components';
import { Container as Link } from '../Link/styles';

export const Container = styled.footer`
  
`;

export const NavLink = styled.div`
  ${({ theme }) => css`
    ${Link} {
      opacity: 0.7;
      &.active {
        color: ${theme.colors.secondaryColor};
        opacity: 1;
      }
    }
  `}
`;

export const NavIcon = styled.div`
  margin: 24px 0;
`;

export const Icon = styled.a`
  padding: 0 1.25rem;
  
`;

export const Text = styled.p`
  ${({ theme }) =>
    css`
      font-size: 0.75rem;
      color: ${theme.colors.colorText};
    `}
`;

export const Content = styled.div`
  border-top: 1px solid #b0e0d3;
  padding-top: 2rem;
  
`;
