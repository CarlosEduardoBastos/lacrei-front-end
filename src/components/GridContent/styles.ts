import styled, { css } from 'styled-components';
import { Container as TextComponent } from '../TextComponent/styles';

const mediaQuery = () => css`
  @media (max-width: 1088px) {
    flex-direction: column;
    align-items: center;
  }
`;

export const Container = styled.div`
  padding-top: 4rem;
`;

export const Grid = styled.div`
  display: flex;
  justify-content: space-between;
  flex-wrap: wrap;
  ${mediaQuery()};
`;
export const ContentText = styled.div`
  ${({ border }:{border: boolean}) => css`
    max-width: 453px;
    ${TextComponent} {
      ${() =>
        border
          ? css`
              border-left: 5px solid rgb(1, 135, 98);
              padding: 0 17px;
            `
          : ''}
    }
  `}
`;

export const ContentImage = styled.div`
  max-width: 540px;
`;
export const Image = styled.img`
  width: 100%;
`;
export const ContentButton = styled.div`
  max-width: 453px;
  display: flex;
  align-items: center;
  justify-content: space-between;
  margin-top: 48px;
  flex-wrap: wrap;
  ${mediaQuery()}
`;
