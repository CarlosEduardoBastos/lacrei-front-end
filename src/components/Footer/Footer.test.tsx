import { Footer } from '.';
import { renderTheme } from '../../styles/render-theme';

describe('<Footer />', () => {
  it('should match snapshot', () => {
    const { container } = renderTheme(<Footer />);
    expect(container).toMatchSnapshot();
  });
});
