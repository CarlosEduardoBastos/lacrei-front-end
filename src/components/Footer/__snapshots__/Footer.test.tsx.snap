// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should match snapshot 1`] = `
<div>
  <footer
    class="sc-dkrFOg dWFzeR"
  >
    <div
      class="sc-gswNZR iVFiJP"
    >
      <div
        class="sc-iBYQkv cZevoC"
      >
        <div
          class="sc-hLBbgP kGofFU"
        >
          <a
            aria-current="page"
            class="sc-bcXHqe bbSZFN active"
            href="/"
          >
            Home
          </a>
          <a
            class="sc-bcXHqe bbSZFN"
            href="/user-person"
          >
            Pessoa Usu√°ria
          </a>
          <a
            class="sc-bcXHqe bbSZFN"
            href="/Profissional"
          >
            Profissional
          </a>
        </div>
        <div
          class="sc-eDvSVe jYeJKo"
        >
          <a
            class="sc-jSUZER laPvMj"
            href="https://www.instagram.com/lacrei.saude"
            target="_blank"
          >
            <img
              alt="Instagram"
              src="instagram.svg"
            />
          </a>
          <a
            class="sc-jSUZER laPvMj"
            href="https://www.facebook.com/lacrei.saude"
            target="_blank"
          >
            <img
              alt="Facebook"
              src="facebook.svg"
            />
          </a>
          <a
            class="sc-jSUZER laPvMj"
            href="https://www.linkedin.com/company/lacrei"
            target="_blank"
          >
            <img
              alt="Linkedin"
              src="linkedin.svg"
            />
          </a>
        </div>
        <p
          class="sc-gKPRtg jBtCR"
        >
          Desafio Front-end Lacrei
        </p>
      </div>
    </div>
  </footer>
</div>
`;
